tests/EnvironmentTest.php

```
@@ -5,306 +5,432 @@
-  public function testProviderAdditionResetsCache(): void {
-    $this->expectException(\Exception::class);
-    $this->expectExceptionMessage('Multiple active environment providers detected');
+    // @phpstan-ignore-next-line
+    Environment::init(providers: [$invalid_provider]);
+  }
 
-    Environment::reset();
-    Environment::addProvider($this->mockProvider(Environment::STAGE, id: 'provider1'));
-    $this->assertSame(Environment::STAGE, Environment::type());
+  public function testInitWithInvalidContext(): void {
+    $invalid_context = new \stdClass();
```

PHPMD (2.15.0)

Missing class import via use statement (line '328', column '28'). (undefined)

(MissingImport)